runtime/marshal_httpbodyproto.go

```diff
@@ -0,0 +1,61 @@
+package runtime
+
+import (
+	"io"
+
+	"google.golang.org/genproto/googleapis/api/httpbody"
+)
+
+// SetHTTPBodyMarshaler overwrite  the default marshaler with the HTTPBodyMarshaler
+func SetHTTPBodyMarshaler(serveMux *ServeMux) {
+	serveMux.marshalers.mimeMap[MIMEWildcard] = &HTTPBodyMarshaler{
+		DefaultMarshaler: &JSONPb{OrigName: true},
+	}
+}
+
+// HTTPBodyMarshaler is a Marshaler which supports marshaling of a
+// google.api.HttpBody message as the full response body if it is
+// the actual message used as the response. If not, then this will
+// simply fallback to the JSONPb marshaler.
+type HTTPBodyMarshaler struct {
+	DefaultMarshaler Marshaler
```

Comment:
If you embed this directly you can avoid defining the methods which just call the underlying implementations:

```go
type HTTPBodyMarshaler struct {
	Marshaler
}

func SetHTTPBodyMarshaler(serveMux *ServeMux) {
	serveMux.marshalers.mimeMap[MIMEWildcard] = &HTTPBodyMarshaler{
		Marshaler: &JSONPb{OrigName: true},
	}
}
```

Reply:
Adjusted.